openedx/core/djangoapps/auth_exchange/tests/test_views.py

```diff
@@ -236,7 +239,11 @@ def test_failure_with_valid_symmetric_jwt(self):
         assert '_auth_user_id' not in self.client.session
         assert response.status_code == 401
 
-    def test_failure_with_valid_asymmetric_jwt_and_disabled_user(self):
+    @mock.patch.object(NotJwtRestrictedApplication, 'has_permission')
+    @mock.patch.object(JwtAuthentication, 'authenticate')
+    def test_failure_with_valid_asymmetric_jwt_and_disabled_user(self, jwt_authenticate, jwt_has_permissions):
+        jwt_authenticate.return_value = (self.user, 'mock-auth')
+        jwt_has_permissions.return_value = True
```

Comment:
I am wondering why we needed to introduce mocking, and what that implies. Hopefully we didn't break any cases, but I don't like that we made testing more complicated. I need to better understand this change, but a synchronous discussion might be better.